### PR TITLE
[risk=no] Reactify inactivity component

### PR DESCRIPTION
**Pull request by calbach**
Move inactivity timeout functionality into a separate React child component, as part of the signed in component reactification.

### CODE REVIEW

**Comment by als364** [comment]
confirmed this DOES work by setting timeout to 60 seconds.

**Comment by als364** [comment]
in case I don't get back to this: it works, but it doesn't handle resetting the timeout well. Dan Abramov suggests [this approach](https://overreacted.io/making-setinterval-declarative-with-react-hooks/)

**Comment by als364** [comment]
With the small changes suggested in the comment, this actually still does work. Agreed it probably makes sense to break this up not as part of this PR, mostly so that we aren't blocking further reactification efforts.

File: ui/src/app/pages/signed-in/inactivity-monitor.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  }
+
+  // Signal user activity.
+  useEffect(() => {
+    let getUserActivityTimer: () => Timeout;
+    let inactivityInterval: Timeout;
+    let logoutTimer: Timeout;
+    let inactivityModalTimer: Timeout;
+
+    function startUserActivityTracker() {
```
Comment (calbach) [inline ui/src/app/pages/signed-in/inactivity-monitor.tsx:63]: nit: this was my laziness in creating the initial version, but most of our code would use `const f = () => {...` style to declare local functions. We should probably have something in our style guide about preference on this as well. There are some subtle differences between these two syntaxes, which are unlikely to matter here, but might matter in other contexts (specifically binding of `this`).

**Review by calbach — COMMENTED** [review]
Looks good

File: ui/src/app/pages/signed-in/inactivity-monitor.tsx (hunk 1 above)
Comment (als364) [reply ui/src/app/pages/signed-in/inactivity-monitor.tsx:63]: Added this to the style guide, and will at least constify the functions I pulled out of the useEffect.

**Review by ericsong — APPROVED** [review]
LGTM

**Pull request merged**